include/sys_clock.h

```diff
@@ -31,40 +33,6 @@ extern int _sys_clock_always_on;
 extern void z_enable_sys_clock(void);
 #endif
 
-#if defined(CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME)
-__syscall int z_clock_hw_cycles_per_sec_runtime_get(void);
-
-static inline int z_impl_z_clock_hw_cycles_per_sec_runtime_get(void)
-{
-	extern int z_clock_hw_cycles_per_sec;
-
-	return z_clock_hw_cycles_per_sec;
-}
-#endif /* CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME */
-
-static inline int sys_clock_hw_cycles_per_sec(void)
```

Comment:
The only thing about this patch that doesn't sit well with me is that we are removing some APIs which have been in public scope:

sys_clock_hw_cycles_per_sec
sys_clock_hw_cycles_per_tick
SYS_CLOCK_HW_CYCLES_TO_NS
SYS_CLOCK_HW_CYCLES_TO_NS64

Could we #define some or all of these in terms of the new APIs and deprecate them, instead of deleting outright?

Reply:
As far as I can tell these are all still here, they've just been moved (and deprecated) within the same header (see down around line 75).